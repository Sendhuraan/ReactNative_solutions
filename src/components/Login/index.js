import React from 'react';
import {View, TextInput, Button} from 'react-native';

import styles from './styles';

function Login(props) {
	const {navigation} = props;

	return (
		<View style={styles.container}>
			<TextInput placeholder="Username" />
			<TextInput placeholder="Password" />
			<Button
				title="Go to Dashboard"
				onPress={() => {
					navigation.navigate('AppTabs');
				}}
			/>
		</View>
	);
}

export default Login;
