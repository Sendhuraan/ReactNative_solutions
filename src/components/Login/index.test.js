import React from 'react';
import Login from '../Login';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
	const tree = renderer.create(<Login />).toJSON();
	expect(tree).toMatchSnapshot();
});
