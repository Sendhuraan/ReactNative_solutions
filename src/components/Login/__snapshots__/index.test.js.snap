// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 50,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    placeholder="Username"
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    placeholder="Password"
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
  />
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Go to Dashboard
      </Text>
    </View>
  </View>
</View>
`;
